lib/timeline/component/Group.js

```diff
@@ -210,19 +210,41 @@ Group.prototype._didMarkerHeightChange = function() {
   var markerHeight = this.dom.marker.clientHeight;
   if (markerHeight != this.lastMarkerHeight) {
     this.lastMarkerHeight = markerHeight;
-    util.forEach(this.items, function (item) {
+    var redrawQueue = {};
+    var redrawQueueLength = 0;
+
+    util.forEach(this.items, function (item, key) {
       item.dirty = true;
-      if (item.displayed) item.redraw();
-    });
+      if (item.displayed) {
+        var returnQueue = true;
+        redrawQueue[key] = item.redraw(returnQueue);
+        redrawQueueLength = redrawQueue[key].length;
+      }
+    })
+
+    if (redrawQueueLength) {
+      // redraw all regular items
+      for (var i = 0; i < redrawQueueLength; i++) {
+        util.forEach(redrawQueue, function (fns) {
+          fns[i]();
+        });
+      }
+    }
+    // util.forEach(this.items, function (item) {
+    //   item.dirty = true;
+    //   if (item.displayed) item.redraw();
+    // });
     return true;
   }
 }
 
 Group.prototype._calculateGroupSizeAndPosition = function() {
-  var foreground = this.dom.foreground;
-  this.top = foreground.offsetTop;
-  this.right = foreground.offsetLeft;
-  this.width = foreground.offsetWidth;
+  var offsetTop = this.dom.foreground.offsetTop
```

Comment:
it's not superfluous.
This avoids the browser from read,write,read,write issues. 
If you put together all the read actions (reading properties from the dom), the browser will get these properties in the next frame while avoiding thrashing (going to read a property, changing the dom, reading another property and changing the dom, and again reading a property and changing the dom - each read and write is a reflow - which is heavy).

Reply:
OK, TIL.

Reply:
[Is this relevat](http://wilsonpage.co.uk/preventing-layout-thrashing/)?

If so, very interesting. This is something that definitely should be checked everywhere in the code, to improve performance.

Reply:
Cool idea. At least initially it might be worth considering replacing the various places `setTimeout` is used for animation frames with calls to `requestAnimationFrame` before going deeper. For some items though (like #3513) the library is relying on the fact that the DOM will get updated in order for it to perform later calculations for things like stacking. This would still work with the idea from the article but would likely require quite a bit of refactoring.